### PR TITLE
HiKey960: fix for 4G boards and support 6G boards

### PR DESCRIPTION
**Pull request by jforissier**

### CODE REVIEW

File: core/arch/arm/plat-hikey/platform_config.h [hunk 1]
```diff
@@ -121,21 +121,33 @@
 
 #elif defined(PLATFORM_FLAVOR_hikey960)
 
+/*
+ * Physical address ranges for HiKey960 RAM:
+ * 3G board: 0~3G
+ * 4G board: 0~3G 3~3.5G 8~8.5G
+ * 4G board: 0~3G 3G~3.5G 8G~8.5G
```
Comment (jbech-linaro) [inline core/arch/arm/plat-hikey/platform_config.h]: This particular line looks a bit confusing, since it's kind of "the same" with the previous line.

Comment (jforissier) [reply core/arch/arm/plat-hikey/platform_config.h]: Doh! Bad conflict resolution on my part as I used git `rebase -i`. Will fix.

File: core/arch/arm/plat-hikey/main.c [hunk 2]
```diff
@@ -55,6 +55,9 @@ register_dynamic_shm(DRAM0_BASE, DRAM0_SIZE_NSEC);
 #ifdef DRAM1_SIZE_NSEC
 register_dynamic_shm(DRAM1_BASE, DRAM1_SIZE_NSEC);
 #endif
+#ifdef DRAM1_SIZE_NSEC
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-hikey/main.c]: DRAM2_SIZE_NSEC?

Comment (jforissier) [reply core/arch/arm/plat-hikey/main.c]: Oops. Yes of course.

**Comment by jforissier** [comment]
Anyone with a 4G board could test this? @taiji76 maybe?

**Comment by taiji76** [comment]
@jforissier 
This patch works very well on my Hikey 960 board.
Thanks a lot!

**Comment by jforissier** [comment]
@taiji76 great. Feel free to add your `Tested-by:` if you want (no obligation of course).

**Comment by taiji76** [comment]
`Tested-by:  Sungjin Park <sungjinp@gmail.com>`

**Comment by mofanv** [comment]
Still had `Bad arg address` error, but fixed by adding
```
CFG_DRAM_SIZE_GB=4
CFG_CORE_LARGE_PHYS_ADDR=y
CFG_ARM64_core=y
```
in `conf.mk` of plat-hikey.

`Tested-by: Fan Mo <f.mo18@imperial.ac.uk>`

**Comment by vchong** [comment]
Just a note that `CFG_ARM64_core` has to be set manually for hikey, which will then set `CFG_WITH_LPAE`, which is basically what setting `CFG_CORE_LARGE_PHYS_ADDR` will do too in `core/arch/arm/arm.mk`. `CFG_DRAM_SIZE_GB=4` has to be set manually too since the default is `3`.

**Comment by liuyunli** [comment]
can you cherry-pick the resolution to 3.8.0 branch @jforissier . As someone may intend to use the stable release branch. Thank you!

**Comment by jforissier** [comment]
@liuyunli there is no `3.8.0` branch, `3.8.0` is a tag on the `master` branch. We don't have "stable release" branches upstream. All fixes get merged into master and are therefore available in the next release (in this case, it will be 3.9.0).
Downstream projects are of course free to maintain their own stable branches and cherry-pick/backport the fixes.

**Pull request merged**